infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/project/OpenShiftProjectFactory.java

```diff
@@ -52,11 +55,21 @@ public OpenShiftProjectFactory(
    * @throws InfrastructureException if any exception occurs during project preparing
    */
   public OpenShiftProject create(String workspaceId) throws InfrastructureException {
-    final String projectName = isNullOrEmpty(this.projectName) ? workspaceId : this.projectName;
+    boolean isPredefined = !isNullOrEmpty(this.projectName);
```

Comment:
Looks like this line can be moved to the constructor

Reply:
Yeap, it even can be moved to the constructor of parent class `KubernetesNamespaceFactory`.

So, really thank you for this comment. Because of that, I believe that I found an issue in `KubernetesNamespaceFactory`. `isPredefined` method returns true if the configured namespace `isNullOrEmpty` and false otherwise, but it should do vice versa.

It may be a cause of appearing a lot of error messages in ci-jobs tests about failed PVC clean up jobs.

I'll check it and will consider adding unit test for `KubernetesNamespaceFactory`.

Reply:
Done.

Also, I added dummy test class for OpenShiftProjectFactory and will fix it before merge.